### PR TITLE
webrtc: Test legacy offerToReceiveAudio/Video options

### PR DESCRIPTION
**Pull request by adam-be**

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/21086)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/307937141)

**Comment by alvestrand** [comment]
I think this won't pass on Chrome (no transceivers), but does it "work" at least to the point where the test is parsed and calls the functions that it should?
What are the results on other browsers?

**Comment by snuggs** [comment]
@alvestrand ref to no transceivers? Or is this wpt related? Would like to get versed.

**Comment by adam-be** [comment]
It won't pass on Chrome (nor Firefox), but from what I see that's the case for all transceiver related tests.

@snuggs, transceivers are part of the spec, but not yet widely implemented.

**Comment by snuggs** [comment]
Copy that @adam-be! That's all I needed to know to start digging.

File: webrtc/RTCPeerConnection-createOffer-offerToReceive.html
Comment (snuggs) [inline webrtc/RTCPeerConnection-createOffer-offerToReceive.html]: @alvestrand what are our conventions for adding author & help meta tags? I have noticed this convention sprinkled throughout the wpt repo (primarily within css subset). I actually wish there were more references especially to the respective spec.

### Examples
  - Spec `help` example: https://github.com/w3c/web-platform-tests/blob/56ea52b8668ceee63ec196c59dd11ad7797e4f0d/html/semantics/embedded-content/the-video-element/video_initially_paused-ref.html#L5
  - `author` example: https://github.com/w3c/web-platform-tests/blob/95eb7be62d2b5a3e9ef1165fd333d353fe3d4b30/html/rendering/bindings/the-button-element/button-type-menu-historical.html#L4

**Review by snuggs — DISMISSED** [review]
Overall well done. Others may want to chime in on this but I encourage a `help` link to the appropriate spec. This allows contributors who may not be explicitly versed in the particular area of the spec. 
This removes the need for questions like following to be answered after PR is merged. https://github.com/w3c/web-platform-tests/pull/8450#issuecomment-347422681

My 2 Satoshi.

**Comment by adam-be** [comment]
Thanks for reviewing. I've added a help link.

I guess with our new "test as you commit" approach there will be situations where a test can't be linked to a section in a publicly available spec since the only place the correct section exists, is in pending PR.

File: webrtc/RTCPeerConnection-createOffer-offerToReceive.html
Comment (alvestrand) [reply webrtc/RTCPeerConnection-createOffer-offerToReceive.html]: No idea about the convention for meta tags, I've not used them; this should have recommendations at the WPT project level - @foolip may want to comment, but that shouldn't block this PR.

**Review by alvestrand — APPROVED** [review]
This looks good to me. I think this tests the existing spec, so should be submitted without waiting for anything. In my opinion, tests that test changes not committed to the spec should be committed when the change to the spec is committed - there may be more opinions on that.

**Comment by adam-be** [comment]
> In my opinion, tests that test changes not committed to the spec should be committed when the change to the spec is committed - there may be more opinions on that.

Sounds reasonable to me.

File: webrtc/RTCPeerConnection-createOffer-offerToReceive.html
Comment (snuggs) [reply webrtc/RTCPeerConnection-createOffer-offerToReceive.html]: >  that shouldn't block this PR @alvestrand 
Agreed

Comment (foolip) [reply webrtc/RTCPeerConnection-createOffer-offerToReceive.html]: You can add it if you find it useful. Myself, I never add author info because it's in git history, and add spec links only if the path and filename doesn't have all the info you need, which is often the case.

**Comment by jan-ivar** [comment]
So I ran this test in Firefox Nightly (which supports transceivers now), and found a [problem](https://github.com/w3c/webrtc-pc/pull/1672#pullrequestreview-80322896): The logic around "stopped" looks flipped.

offerToReceiveAudio option should be ignored if a stopped "sendrecv" transceiver exists
---------------------------------------------------------------------------------------
Expected PASS, got FAIL
assert_equals: Expect pc to still have one transceiver expected 1 but got 2
@http://web-platform.test:8000/webrtc/RTCPeerConnection-createOffer-offerToReceive.html:60:9
promise callback*@http://web-platform.test:8000/webrtc/RTCPeerConnection-createOffer-offerToReceive.html:58:14
Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1489:20
promise callback*promise_test@http://web-platform.test:8000/resources/testharness.js:539:31
@http://web-platform.test:8000/webrtc/RTCPeerConnection-createOffer-offerToReceive.html:46:5
@http://web-platform.test:8000/webrtc/RTCPeerConnection-createOffer-offerToReceive.html:11:3

offerToReceiveAudio option should be ignored if a stopped "recvonly" transceiver exists
---------------------------------------------------------------------------------------
Expected PASS, got FAIL
assert_equals: Expect pc to still have one transceiver expected 1 but got 2
@http://web-platform.test:8000/webrtc/RTCPeerConnection-createOffer-offerToReceive.html:80:9
promise callback*@http://web-platform.test:8000/webrtc/RTCPeerConnection-createOffer-offerToReceive.html:78:14
Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1489:20
promise callback*promise_test@http://web-platform.test:8000/resources/testharness.js:539:31
@http://web-platform.test:8000/webrtc/RTCPeerConnection-createOffer-offerToReceive.html:65:5
@http://web-platform.test:8000/webrtc/RTCPeerConnection-createOffer-offerToReceive.html:11:3

offerToReceiveVideo option should be ignored if a stopped "sendrecv" transceiver exists
---------------------------------------------------------------------------------------
Expected PASS, got FAIL
assert_equals: Expect pc to still have one transceiver expected 1 but got 2
@http://web-platform.test:8000/webrtc/RTCPeerConnection-createOffer-offerToReceive.html:60:9
promise callback*@http://web-platform.test:8000/webrtc/RTCPeerConnection-createOffer-offerToReceive.html:58:14
Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1489:20
promise_test/tests.promise_tests<@http://web-platform.test:8000/resources/testharness.js:543:27
promise callback*promise_test@http://web-platform.test:8000/resources/testharness.js:539:31
@http://web-platform.test:8000/webrtc/RTCPeerConnection-createOffer-offerToReceive.html:46:5
@http://web-platform.test:8000/webrtc/RTCPeerConnection-createOffer-offerToReceive.html:11:3

offerToReceiveVideo option should be ignored if a stopped "recvonly" transceiver exists
---------------------------------------------------------------------------------------
Expected PASS, got FAIL
assert_equals: Expect pc to still have one transceiver expected 1 but got 2
@http://web-platform.test:8000/webrtc/RTCPeerConnection-createOffer-offerToReceive.html:80:9
promise callback*@http://web-platform.test:8000/webrtc/RTCPeerConnection-createOffer-offerToReceive.html:78:14
Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1489:20
promise_test/tests.promise_tests<@http://web-platform.test:8000/resources/testharness.js:543:27
promise callback*promise_test@http://web-platform.test:8000/resources/testharness.js:539:31
@http://web-platform.test:8000/webrtc/RTCPeerConnection-createOffer-offerToReceive.html:65:5
@http://web-platform.test:8000/webrtc/RTCPeerConnection-createOffer-offerToReceive.html:11:3

**Comment by adam-be** [comment]
@jan-ivar, that was my mistake. I've updated the test now and it's all green in Firefox nightly. :)

**Comment by adam-be** [comment]
@snuggs, do you believe that your comments above have been addressed?

File: webrtc/RTCPeerConnection-createOffer-offerToReceive.html
Comment (fippo) [inline webrtc/RTCPeerConnection-createOffer-offerToReceive.html]: whitespace nit: indent 2 spaces?
can fix when adding tests for https://github.com/w3c/webrtc-pc/pull/1693

Comment (adam-be) [reply webrtc/RTCPeerConnection-createOffer-offerToReceive.html]: Fixed.

**Comment by adam-be** [comment]
It would be great to merge this PR since we want to base additional tests on it. @snuggs and @foolip, are you happy with the current state of the PR?

File: webrtc/RTCPeerConnection-createOffer-offerToReceive.html
Comment (fippo) [inline webrtc/RTCPeerConnection-createOffer-offerToReceive.html]: missing return?

Comment (adam-be) [reply webrtc/RTCPeerConnection-createOffer-offerToReceive.html]: True; will fix.

**Comment by alvestrand** [comment]
@snuggs your review is blocking this merge, and was (I think) addressed ~20 days ago. Can you unblock?

**Review by fippo — APPROVED** [review]

**Pull request merged**
